pydantic/fields.py

```diff
@@ -238,8 +245,11 @@ def validate(self, v, values, *, loc, cls=None):
         if self.allow_none and v is None:
             return None, None
 
-        if not isinstance(loc, tuple):
-            loc = (loc,)
+        loc = loc if isinstance(loc, tuple) else (loc, )
+
+        v, error = self._validate_json(v, loc)
```

Comment:
Please can you move the `if self.parse_json` to here.

so

```py
if self.parse_json:
    v, error = self._validate_json(v, loc)
    if error:
        ...
```

Reply:
I moved checking of `self.parse_json` to `self._validate_json` in order to skip from flake8 warning `pydantic/fields.py:244:5: C901 'Field.validate' is too complex (11)`

Reply:
Thought that might be the case, just add `  # noqa: C901 (ignore complexity)` to the end of the function definition.